### PR TITLE
source-mssql: shorten capture job polling interval in tests

### PR DESCRIPTION
**Pull request by postamar**
This PR should reduce flakiness in tests.

SQL Server is different than Postgres and MySql in that its CDC functionality consists of more than just parsing the WAL, instead it involves background jobs writing to _CT tables whose contents are eventually consistent. The background job of type "capture" does its thing every 5 seconds by default. If we have two consecutive `read`s within less than 5 seconds we might end up with the same target LSN despite having committed transactions between the two reads!

This PR reduces the polling interval to every 1 second in tests and waits the same amount of time before querying the target LSN. The latter depends on the contents of the _CT tables and we want to give them a chance to be updated.

### CODE REVIEW

**Comment by postamar** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #33510** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/33510?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #33511** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/33511?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/33510?utm_source=stack-comment).

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcTargetPosition.java [hunk 1]
```diff
@@ -77,17 +81,30 @@ public int hashCode() {
...
     try {
-      final List<JsonNode> jsonNodes = database
-          .bufferedResultSetQuery(connection -> connection.createStatement().executeQuery(
-              "USE [" + dbName + "]; SELECT sys.fn_cdc_get_max_lsn() AS max_lsn;"), JdbcUtils.getDefaultSourceOperations()::rowToJson);
+      // We might have to wait a bit before querying the max_lsn to give the CDC capture job
+      // a chance to catch up. This is important in tests, where reads might occur in quick succession
+      // which might leave the CT tables (which Debezium consumes) in a stale state.
+      final JsonNode sourceConfig = database.getSourceConfig();
+      final Duration delay = (sourceConfig != null && sourceConfig.has("is_test") && sourceConfig.get("is_test").asBoolean())
+          ? MAX_LSN_QUERY_DELAY_TEST
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcTargetPosition.java:89]: maybe we should provide that as a parameter to set when creating the source. Different customers might have that set to different values on their source DB

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcTargetPosition.java [hunk 2]
```diff
@@ -77,17 +81,30 @@ public int hashCode() {
...
+          : MAX_LSN_QUERY_DELAY;
+      final String maxLsnQuery = """
+                                 USE [%s];
+                                 WAITFOR DELAY '%02d:%02d:%02d';
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcTargetPosition.java:93]: oh this is ugly! who had the genius idea of forcing a HH:MM:SS format here ???

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcTargetPosition.java (hunk 1 above)
Comment (stephane-airbyte) [reply airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcTargetPosition.java:89]: if you agree, we should just create a github ticket. I'm not asking for that to be part of this change

**Review by stephane-airbyte — APPROVED** [review]

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcTargetPosition.java (hunk 2 above)
Comment (postamar) [reply airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcTargetPosition.java:93]: Bill Gates, most likely. Or someone who doesn't care about ISO 8601.

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcTargetPosition.java (hunk 1 above)
Comment (postamar) [reply airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcTargetPosition.java:89]: I'd like to put this off until we see some actual production usage. The default interval is 5 seconds. Presumably, airbyte syncs don't run that frequently, in which case this is irrelevant. Let's not anticipate problems which may never materialize.

**Comment by postamar** [comment]
This makes the tests flaky for some as-of-yet unknown reason.

**Comment by postamar** [comment]
Fixed. As it turns out, adding the missing `is_test` field in the config does have an impact!

**Pull request merged**